src/encoder.rs

```diff
@@ -162,7 +162,7 @@ fn encode_bson<W: Write + ?Sized>(writer: &mut W, key: &str, val: &Bson) -> Enco
             try!(writer.write_u8(From::from(subtype)));
             writer.write_all(data).map_err(From::from)
         },
-        &Bson::UtcDatetime(ref v) => write_i64(writer, v.timestamp()),
+        &Bson::UtcDatetime(ref v) => write_i64(writer, v.timestamp() * 1000),
```

Comment:
`v.nanoseconds()` doesn't seem to exist, There is a `nanosecond()` trait in the `TimeLike` trait (https://lifthrasiir.github.io/rust-chrono/chrono/trait.Timelike.html#method.nanosecond) 

If I understand the docs correctly it should become: 

```rust
        &Bson::UtcDatetime(ref v) => {
            write_i64(writer, ((v.timestamp() * 1000) + (v.nanosecond()  / 1000000)  as i64))
        },
```